packages/flutter/lib/src/widgets/framework.dart

```
@@ -159,46 +172,83 @@ abstract class GlobalKey<T extends State<StatefulWidget>> extends Key {
+          // _debugVerifyIllFatedPopulation or
+          // _debugElementsThatWillNeedToBeRebuiltDueToGlobalKeyShenanigans.
+          if (keyToParent.containsKey(key) && keyToParent[key] != parent) {
+            // We have duplication reservations for the same global key.
+            final Element older = keyToParent[key];
+            final Element newer = parent;
+            FlutterError error;
+            if (older.toString() != newer.toString()) {
+              error = FlutterError.fromParts(<DiagnosticsNode>[
+                ErrorSummary('Multiple widgets used the same GlobalKey.'),
+                ErrorDescription(
+                  'The key $key was used by multiple widgets. The parents of those widgets were:\n'
+                    '- ${older.toString()}\n'
+                    '- ${newer.toString()}\n'
+                    'A GlobalKey can only be specified on one widget at a time in the widget tree.'
+                ),
+              ]);
+            } else {
+              error = FlutterError.fromParts(<DiagnosticsNode>[
+                ErrorSummary('Multiple widgets used the same GlobalKey.'),
+                ErrorDescription(
+                  'The key $key was used by multiple widgets. The parents of those widgets were '
+                    'different widgets that both had the following description:\n'
+                    '  ${parent.toString()}\n'
+                    'A GlobalKey can only be specified on one widget at a time in the widget tree.'
+                ),
+              ]);
+            }
+            // A child can attach to two parents due to global key duplication
+            // between MultiChildRenderObjectElement and its children subtrees.
+            try {
+              if (child._parent != older)
+                older.forgetChild(child);
+              if (child._parent != newer)
+                newer.forgetChild(child);
+            } on Error {
```

Can we be more specific here? If it's any unrelated error we do want to see that one and not silently swallow it!